### PR TITLE
refactor(streaming): ensure correct storage table constructing for all streaming operators 

### PR DESCRIPTION
**Pull request by wcy-fdu**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?
As title.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 
* Connector (sources & sinks)
* SQL commands, functions, and operators
* RisingWave cluster configuration changes
* Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

## Refer to a related PR or issue link (optional)

close https://github.com/singularity-data/risingwave/issues/3753

### CODE REVIEW

**Comment by wcy-fdu** [comment]
Maybe the `vnode_bitmap` for streaming look_up is incorrect. And idea?cc  @BowenXiao1999  @BugenZhao 
image

**Comment by wcy-fdu** [comment]
> What about try `arrange_key` (the join key) first? Although the dist key should be determined by frontend and may not be join key... But I think it can cover most cases.
> 
> Given that the lookup of state table is not from Frontend yet. I'll check whether it is doable.
> 
> cc @BugenZhao

Still fail, I will take a look later.

File: src/storage/src/table/state_table.rs [hunk 1]
```diff
@@ -57,9 +57,8 @@ pub struct StateTableBase<S: StateStore, E: Encoding> {
 }
 
 impl<S: StateStore, E: Encoding> StateTableBase<S, E> {
-    /// Note: `dist_key_indices` is ignored, use `new_with[out]_distribution` instead.
-    // TODO: remove this after all state table usages are replaced by `new_with[out]_distribution`.
-    pub fn new(
+    /// Create a state table without distribution, used for tests.
+    pub fn new_for_test(
```
Comment (BugenZhao) [inline src/storage/src/table/state_table.rs:61]: Let's remove this and replace all references with `new_without_distribution`?

**Review by BugenZhao — COMMENTED** [review]
LGTM for the state table. Not sure whether the dist key of `Lookup` is correct. 😄 cc @skyzh

File: src/storage/src/table/state_table.rs (hunk 1 above)
Comment (wcy-fdu) [reply src/storage/src/table/state_table.rs:61]: LGTM

**Comment by BowenXiao1999** [comment]
Hacked a bit and I found that the distribution key of ArrangeNode do not include join key (seems by design). BTW if we want to enable VNode for Lookup, we should enable VNode for MView first

**Comment by skyzh** [comment]
related: https://github.com/singularity-data/risingwave/issues/3796. We will resolve this later.

**Review by skyzh — APPROVED** [review]
LGTM for the delta join part. Need reviews from other members.

**Pull request merged**